scripts/import_from_ena_legacy.py

```diff
@@ -0,0 +1,849 @@
+                tech_type.setdefault(TECHNOLOGIES[assay_type], 0)
+                tech_type[TECHNOLOGIES[assay_type]] += 1
+                experiment_type.setdefault(assay_type, 0)
+                experiment_type[assay_type] += 1
+            else:
+                pass
+        num_valid_exps = len(only_valid_exps.keys())
+        if num_valid_exps == 0:
+            logger.warning(f"dataset {dataset_id} has no valid experiments, skipped.")
+            continue
+        es_doc_dataset['standardMet'] = dataset_standard
+        specimen_set = datasets['tmp'][dataset_id]['specimen']
+        species = dict()
+        specimens_list = list()
+        for specimen in specimen_set:
+            if specimen not in BIOSAMPLES_RECORDS:
+                logger.warning(f"BioSamples record {specimen} required by dataset {dataset_id} could not be found")
+                continue
+            specimen_detail = BIOSAMPLES_RECORDS[specimen]
+            es_doc_specimen = dict()
+            es_doc_specimen['biosampleId'] = check_existsence(specimen_detail, 'biosampleId')
+            es_doc_specimen['material'] = check_existsence(specimen_detail, 'material')
+            es_doc_specimen['cellType'] = check_existsence(specimen_detail, 'cellType')
+            es_doc_specimen['organism'] = check_existsence(specimen_detail['organism'], 'organism')
+            es_doc_specimen['sex'] = check_existsence(specimen_detail['organism'], 'sex')
+            es_doc_specimen['breed'] = check_existsence(specimen_detail['organism'], 'breed')
+            specimens_list.append(es_doc_specimen)
+            species[specimen_detail['organism']['organism']['text']] = specimen_detail['organism']['organism']
+        if not specimens_list:  # no sepcimen found for the dataset, skip
+            logger.warning(f"Dataset {dataset_id} could not find any related specimen, skipped")
+            continue
+        es_doc_dataset['specimen'] = sorted(specimens_list, key=lambda k: k['biosampleId'])
+        es_doc_dataset['species'] = list(species.values())
+        file_arr = datasets['tmp'][dataset_id]['file'].values()
+        valid_files = list()
+        for file_entry in sorted(file_arr, key=lambda k: k['name']):
+            file_id = file_entry['fileId']
+            if file_id in indexed_files:
+                valid_files.append(file_entry)
+        es_doc_dataset['file'] = valid_files
+        es_doc_dataset['experiment'] = list(only_valid_exps.values())
+        es_doc_dataset['assayType'] = list(experiment_type.keys())
+        es_doc_dataset['tech'] = list(tech_type.keys())
+        es_doc_dataset['instrument'] = list(datasets['tmp'][dataset_id]['instrument'])
+        es_doc_dataset['archive'] = sorted(list(datasets['tmp'][dataset_id]['archive']))
+        body = json.dumps(es_doc_dataset)
+        utils.insert_into_es(es, es_index_prefix, 'dataset', dataset_id, body)
+    logger.info("finishing indexing datasets")
+
+def get_ena_data():
```

Comment:
We already have this function in import_from_ena script, so it should be extracted to misc module

Reply:
removed in 27a573a